src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/mainscreen/view/MainScreenView.kt

```diff
@@ -53,6 +57,19 @@ class MainScreenView : View() {
                                     titleProperty.bind(viewModel.selectedCollectionTitle)
                                     bodyTextProperty.bind(viewModel.selectedCollectionBody)
                                     visibleProperty().bind(viewModel.selectedCollectionProperty.booleanBinding { it != null })
+                                    showProgress = true
+                                    //listen for changes to the active collection to update the progress
+                                    viewModel.selectedCollectionProperty.onChange { collection ->
+                                        if(collection != null) {
+                                        progressProperty.bind(viewModel.getProgress(collection))
+                                        }
+                                        else progressProperty.unbind()
+                                    }
+
+                                    viewModel.selectedTake.onChange {//listen for selected take changes to update progress
+                                        progressProperty.unbind()
+                                        progressProperty.bind(viewModel.getProgress(viewModel.selectedCollectionProperty.value))
```

Comment:
This is a little picky, but I'd create a private function to bind `progressProperty` since you also do the same thing at line 64. Also, I guess there's no way to avoid doing this unbind() and rebind? It's too bad `getProgress` doesn't know when a change to it's output value would occur but obviously that's impossible lol.

Reply:
Not impossible.  Can use [createDoubleBinding](https://docs.oracle.com/javase/8/javafx/api/javafx/beans/binding/Bindings.html#createDoubleBinding-java.util.concurrent.Callable-javafx.beans.Observable...-) and give selectedTake as one of the dependencies.

Reply:
Aha, awesome

Reply:
@natibekele perhaps this would work then
```
progressProperty.bind(
	Bindings.createDoubleBinding(
		Callable {
			viewModel.getProgress(collection)
		},
		viewModel.selectedTake
	)
)
```